### PR TITLE
Refactor testconn to use get_sqla_engine

### PR DESCRIPTION
**Pull request by thoralf-gutierrez**
### CATEGORY

Choose one

- [ ] Bug Fix
- [ ] Enhancement (new features, refinement)
- [x] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY
Creating the database connection will actually go through the get_sqla_engine path so that's what should be used to test the connection. This was triggered by realizing that testconn was not using the DB_CONNECTION_MUTATOR hook.

See https://github.com/apache/incubator-superset/issues/7668

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/views/core.py [hunk 1]
```diff
@@ -1796,46 +1795,29 @@ def testconn(self):
             uri = request.json.get('uri')
             db_name = request.json.get('name')
             impersonate_user = request.json.get('impersonate_user')
-            database = None
+
+            # extras is sent as json, but required to be a string in the Database model
+            extra_json = request.json.get('extras', {})
```
Comment (john-bodley) [inline superset/views/core.py]: I would probably just make this one line called `extras` (inline with the JSON name), i.e., 

```
extras = json.dumps(request.json.get('extras', {}))
```

File: superset/views/core.py [hunk 2]
```diff
@@ -1796,46 +1795,29 @@ def testconn(self):
...
+            extra_str = json.dumps(extra_json)
+
             if db_name:
-                database = (
+                existing_database = (
```
Comment (john-bodley) [inline superset/views/core.py]: Would you be able to clarify the difference between `existing_database` and `database` (defined below)?

**Review by john-bodley — COMMENTED** [review]
Thanks for refactoring the code.

**Comment by thoralf-gutierrez** [comment]
Thanks for the early comments @john-bodley -- I fixed how I was creating the database (needed to use `set_sqlalchemy_uri`) and now the test passes. I think I've also addressed your comments. Can you take a second look?

File: superset/views/core.py [hunk 3]
```diff
@@ -2018,53 +2017,38 @@ def testconn(self):
...
+                    uri = existing_database.sqlalchemy_uri_decrypted
 
-            engine_params = request.json.get("extras", {}).get("engine_params", {})
-            connect_args = engine_params.get("connect_args")
-
-            if configuration and connect_args is not None:
-                connect_args["configuration"] = configuration
+            # this is the database instance that will be tested
+            database = models.Database(
+                extra=extra_str, impersonate_user=impersonate_user
```
Comment (john-bodley) [inline superset/views/core.py]: Given that `extra_str` isn't used elsewhere I would probably just have all the content/context in one place, 

```
database = models.Database(
    extra=json.dumps(request.json.get("extras", {})), 
    impersonate_user=impersonate_user,
)
```

Comment (thoralf-gutierrez) [reply superset/views/core.py]: :+1: Also went ahead and did the same for other variables.

**Review by john-bodley — APPROVED** [review]

**Comment by john-bodley** [comment]
@thoralf-gutierrez would you mind removing the `[WIP]` prefix if this PR is no longer work-in-progress?

**Review by mistercrunch — APPROVED** [review]

**Pull request merged**